internal/resources/kustomization/resource_kustomization.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package kustomization
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	commonscope "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kustomization/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kustomization/spec"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kustomization/status"
+)
+
+func ResourceKustomization() *schema.Resource {
+	return &schema.Resource{
+		Schema:        kustomizationSchema,
+		CustomizeDiff: schema.CustomizeDiffFunc(commonscope.ValidateScope([]string{commonscope.ClusterKey, commonscope.ClusterGroupKey})),
+	}
+}
+
+var kustomizationSchema = map[string]*schema.Schema{
+	nameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the Kustomization.",
+		Required:    true,
+		ForceNew:    true,
+	},
+	namespaceNameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of Namespace.",
+		Required:    true,
+		ForceNew:    true,
+	},
+	orgIDKey: {
+		Type:        schema.TypeString,
+		Description: "ID of Organization.",
+		Optional:    true,
+		ForceNew:    true,
+		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
explain the context where this will be the case ?

Reply:
This is required because the org id is an optional parameter in the scope/fullname unlike the others, when the user does not provide its value, we get it from the backend in the response and the state file gets updated with the org id.

This causes a diff in each `apply`. So we have put this check to suppress the diff if the org id in the state file is the same as the one coming from the backend.

To add to the above, we cannot make this value as computed since this is something that the client sends and is configurable as per the API spec. It may not be good to mark this field as computed.

Reply:
As discussed, I have removed the `org_id` attribute from the schema since the resources are scoped to `cluster` and `cluster_group`.